app/src/main/java/org/stepic/droid/code/ui/CodeEditorLayout.kt

```diff
@@ -0,0 +1,55 @@
+package org.stepic.droid.code.ui
+
+import android.content.Context
+import android.graphics.Typeface
+import android.util.AttributeSet
+import android.view.LayoutInflater
+import android.widget.ScrollView
+import org.stepic.droid.R
+import org.stepic.droid.code.highlight.themes.CodeTheme
+import org.stepic.droid.code.highlight.themes.Presets
+import org.stepic.droid.util.insertText
+
+class CodeEditorLayout : ScrollView {
+    private val codeEditor: CodeEditor
+
+    var theme : CodeTheme
+        get() = codeEditor.theme
+        set(value) {
+            setBackgroundColor(value.background)
+            codeEditor.theme = value
+        }
+
+    val text : CharSequence
+        get() = codeEditor.text.toString()
+
+    constructor(context: Context) : this(context, null)
+
+    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs) {
```

Comment:
We can use @JvmOverload and default args in constructor (see `LoadingView`). I think it is more elegant solution for custom views